src/service.py

```diff
@@ -49,15 +51,21 @@ def configure(self, snap_config: dict[str, Any]) -> None:
         self.snap_client.set(snap_config, typed=True)
 
 
-def snap_install(resource: str) -> SnapService:
+def snap_install(resource: Optional[str]) -> SnapService:
```

Comment:
If a user attaches a snap resource after deployment, will the snap resource be installed, replacing the snap from the store?  Conversely, if a user removes the snap resource, will the snap revert back to the one from the store?  I think this would be desired behaviour.

Reply:
I took quite some time trying to figure out how I could detect that a snap was installed from a local file and not from snapstore. Long story short there is no clear information about this. The only thing that I found is that when you install locally it will show an  `x` on the revision. This [post](https://forum.snapcraft.io/t/how-to-check-if-a-snap-is-installed-in-dangerous-mode/3968) confirms that.

It's not possible to remove a snap resource from the charm, what you can do is to pass a resource. When that is done, it was failing because the snap couldn't refresh with a local installation. In order to solve this I'm detecting if there is `x` in the version and in that case I remove the snap to be able to install from the snapstore again.

Reply:
thanks for digging into this! :)